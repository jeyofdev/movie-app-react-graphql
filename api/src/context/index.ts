import PersonService from '../services/Person.service';
import CollectionService from '../services/Collection.service';
import MovieService from '../services/Movie.service';
import KeywordService from '../services/Keyword.service';
import GenreService from '../services/Genre.service';
import SearchService from '../services/Search.service';

export interface IContext {
	dataSource: {
		movies: MovieService;
		collections: CollectionService;
		persons: PersonService;
		keywords: KeywordService;
		genres: GenreService;
		search: SearchService;
	};
}

export const context: IContext = {
	dataSource: {
		movies: new MovieService(),
		collections: new CollectionService(),
		persons: new PersonService(),
		keywords: new KeywordService(),
		genres: new GenreService(),
		search: new SearchService(),
	},
};
