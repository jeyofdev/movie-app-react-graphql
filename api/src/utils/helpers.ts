import request from 'supertest';
import { OptionsInput } from '../__generated__/resolvers-types';

export const formatUrlQuery = (
	baseUrl: string,
	apiKey: string,
	pathParams?: string | number,
	queryString?: OptionsInput,
) => {
	let url = `${baseUrl}/`;
	url += pathParams ?? `${pathParams}`;
	url += `?api_key=${apiKey}&`;

	if (queryString) {
		const allQueryStrings = Object.entries(queryString)
			// eslint-disable-next-line @typescript-eslint/no-unused-vars
			.filter(([_, value]) => value)
			.map(value => value.join('='));

		url += allQueryStrings.join('&');
	}
	return url;
};

export const executeRequestForTesting = async (
	url: string,
	query: string,
	variables: unknown,
	statusCode = 200,
) => {
	const response = await request(url)
		.post('/')
		.set('Accept', 'application/json')
		.send({
			query,
			variables,
		})
		.expect(statusCode);

	return response;
};
