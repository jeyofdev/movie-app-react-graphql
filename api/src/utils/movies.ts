import { MoviesResponse } from '../__generated__/resolvers-types';

const setMoviesResult = (movies: MoviesResponse) => ({
	page: movies.page,
	results: movies.results,
	total_pages: movies.total_pages,
	total_results: movies.total_results,
});

export default setMoviesResult;
