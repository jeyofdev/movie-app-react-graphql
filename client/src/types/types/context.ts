import { LanguageEnum } from '@graphql/__generated__/graphql-type';
import { DarkModeEnum } from '../enums';

export type ThemeContextType = {
	themeMode: DarkModeEnum;
	handleThemeMode: () => void;
};

export type TranslationContextType = {
	currentLocale: LanguageEnum;
	handleLocale: (locale: LanguageEnum) => void;
};
