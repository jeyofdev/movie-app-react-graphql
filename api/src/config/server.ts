import { startStandaloneServer } from '@apollo/server/standalone';
import { readFileSync } from 'fs';
import { ApolloServer } from '@apollo/server';
import { IContext, context } from '../context';
import resolvers from '../resolvers';

const getServer = async (port?: number) => {
	const typeDefs = readFileSync('./src/schema/schema.graphql', {
		encoding: 'utf-8',
	});

	const server = new ApolloServer<IContext>({
		typeDefs,
		resolvers,
	});

	const { url } = await startStandaloneServer(server, {
		context: async () => context,
		listen: { port },
	});

	return { server, url };
};

export default getServer;
