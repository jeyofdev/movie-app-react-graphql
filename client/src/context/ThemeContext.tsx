import { createContext, useState } from 'react';
import { DarkModeEnum } from '../types/enums';
import { ThemeContextType } from '../types/types/context';
import { ThemeContextProviderPropsType } from '../types/types/props';

export const ThemeContext = createContext<ThemeContextType>({
	themeMode: DarkModeEnum.DARK,
	handleThemeMode: () => {},
});

const ThemeContextProvider = ({ children }: ThemeContextProviderPropsType) => {
	const [themeMode, setThemeMode] = useState<DarkModeEnum>(DarkModeEnum.DARK);

	const handleThemeMode = () => {
		setThemeMode(
			themeMode === DarkModeEnum.DARK ? DarkModeEnum.LIGHT : DarkModeEnum.DARK,
		);
	};

	return (
		<ThemeContext.Provider value={{ themeMode, handleThemeMode }}>
			{children}
		</ThemeContext.Provider>
	);
};

export default ThemeContextProvider;
