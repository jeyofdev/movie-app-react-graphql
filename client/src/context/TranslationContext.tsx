import { LanguageEnum } from '@graphql/__generated__/graphql-type';
import { createContext, useState } from 'react';
import { TranslationContextType } from '../types/types/context';
import { ThemeContextProviderPropsType } from '../types/types/props';

export const TranslationContext = createContext<TranslationContextType>({
	currentLocale: LanguageEnum.En,
	handleLocale: () => {},
});

const TranslationContextProvider = ({
	children,
}: ThemeContextProviderPropsType) => {
	const [currentLocale, setcurrentLocale] = useState<LanguageEnum>(
		(localStorage.getItem('locale') as LanguageEnum) ?? LanguageEnum.En,
	);

	const handleLocale = (locale: LanguageEnum) => {
		setcurrentLocale(locale);
	};

	return (
		<TranslationContext.Provider value={{ currentLocale, handleLocale }}>
			{children}
		</TranslationContext.Provider>
	);
};

export default TranslationContextProvider;
