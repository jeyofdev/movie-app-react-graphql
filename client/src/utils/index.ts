import { format } from 'date-fns';
import { enGB, fr, es, de } from 'date-fns/locale';
import { toast } from 'react-toastify';
import { LanguageEnum } from '@graphql/__generated__/graphql-type';
import { RoutesEnum } from '../types/enums';

/**
 * Limit text by number words
 * @param text
 * @param limit
 * @returns
 */
export const truncate = (text: string, limit: number): string => {
	const { length } = text.split(' ');

	return length <= limit
		? text
		: `${text.split(' ').slice(0, limit).join(' ')}...`;
};

/**
 * Format number to hours and minutes
 * @param totalMinutes
 * @returns
 */
export const formatNumberToHours = (totalMinutes: number): string => {
	const hours = Math.floor(totalMinutes / 60);
	const minutes = totalMinutes % 60;

	if (minutes === 0) {
		return `${hours}h`;
	}

	if (minutes < 10) {
		return `${hours}h ${`0${minutes}`}min`;
	}

	return `${hours}h ${minutes}min`;
};

export const formatGenreForUrl = (genre: string) =>
	`/${RoutesEnum.MOVIES_GENRE}/${genre.toLowerCase().split(' ').join('-')}`;

export const firstLetterCapitalize = (text: string): string =>
	text.slice(0, 1).toUpperCase() + text.slice(1);

export const getAge = (birthday: string) => {
	const date = new Date(birthday);
	const diff = Date.now() - date.getTime();
	const age = new Date(diff);

	return Math.abs(age.getUTCFullYear() - 1970);
};

export const getAgeBetweenTwoDate = (birthday: string, deathday: string) => {
	const dateBirthday = new Date(birthday);
	const dateDeathday = new Date(deathday);
	const diff = dateDeathday.getTime() - dateBirthday.getTime();
	const age = new Date(diff);

	return Math.abs(age.getUTCFullYear() - 1970);
};

export const formatDate = (date: string, locale: LanguageEnum) => {
	const newDate = new Date(date);

	const formatByLocale = () => {
		switch (locale) {
			case LanguageEnum.En:
				return 'MMMM dd, yyyy';
			case LanguageEnum.Fr:
				return 'dd MMMM yyyy';
			case LanguageEnum.Es:
				return 'dd MMMM yyy';
			case LanguageEnum.De:
				return 'dd. MMMM yyyy';

			default:
				return 'MMMM dd, yyyy';
		}
	};

	const getLocale = () => {
		switch (locale) {
			case LanguageEnum.En:
				return enGB;
			case LanguageEnum.Fr:
				return fr;
			case LanguageEnum.Es:
				return es;
			case LanguageEnum.De:
				return de;

			default:
				return enGB;
		}
	};

	return format(newDate, formatByLocale(), { locale: getLocale() });
};

export const toastSuccess = (message: string, autoClose?: number) =>
	toast.success(message, {
		position: 'top-right',
		autoClose: autoClose ?? 5000,
		hideProgressBar: false,
		closeOnClick: true,
		pauseOnHover: true,
		draggable: true,
		progress: undefined,
		theme: 'colored',
	});
