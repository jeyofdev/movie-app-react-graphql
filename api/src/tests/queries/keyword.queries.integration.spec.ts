import { afterAll, beforeAll, describe, expect, it } from 'vitest';
import { ApolloServer } from '@apollo/server';
import { IContext } from '../../context';
import getServer from '../../config/server';
import 'dotenv/config';
import { executeRequestForTesting } from '../../utils/helpers';
import keywordsQueryOperations from '../../operations/keyword.query.operations';

describe('init server', () => {
	let server: ApolloServer<IContext>;
	let url: string;

	beforeAll(async () => {
		if (!process.env.TEST_API_PORT) {
			throw new Error(
				'The environment variable TEST_API_PORT must be specified',
			);
		}
		({ server, url } = await getServer());
	});

	afterAll(async () => {
		await server.stop();
	});

	describe('Queries collections', () => {
		it('When no keyword found', async () => {
			const response = await executeRequestForTesting(
				url,
				keywordsQueryOperations.moviesByKeyword,
				{
					keywordId: 755,
				},
			);

			expect(response.error).not.toBeUndefined();
			expect(response.body.data.moviesByKeyword).toBeNull();
		});

		it('Returns the list of movies in a collection', async () => {
			const response = await executeRequestForTesting(
				url,
				keywordsQueryOperations.moviesByKeyword,
				{
					keywordId: 161176,
				},
			);

			expect(response.error).not.toBeUndefined();
			expect(response.body.data.moviesByKeyword).toMatchSnapshot();
		});
	});
});
