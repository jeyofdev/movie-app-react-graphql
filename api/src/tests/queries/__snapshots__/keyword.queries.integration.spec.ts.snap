// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`init server > Queries keywords > Returns the list of movies to a keyword 1`] = `
{
  "id": 161176,
  "page": 1,
  "results": [
    {
      "adult": false,
      "backdrop_path": "/51vreLbrGwzAg4WRCHgitWz6Naw.jpg",
      "genre_ids": [
        878,
        12,
        28,
      ],
      "id": 447365,
      "original_language": "en",
      "original_title": "Guardians of the Galaxy Vol. 3",
      "overview": "Peter Quill, still reeling from the loss of Gamora, must rally his team around him to defend the universe along with protecting one of their own. A mission that, if not completed successfully, could quite possibly lead to the end of the Guardians as we know them.",
      "popularity": 1454.395,
      "poster_path": "/r2J02Z2OpNTctfOSN1Ydgii51I3.jpg",
      "release_date": "2023-05-03",
      "title": "Guardians of the Galaxy Vol. 3",
      "video": false,
      "vote_average": 8.1,
      "vote_count": 1579,
    },
    {
      "adult": false,
      "backdrop_path": "/eNJhWy7xFzR74SYaSJHqJZuroDm.jpg",
      "genre_ids": [
        28,
        878,
      ],
      "id": 1033219,
      "original_language": "en",
      "original_title": "Attack on Titan",
      "overview": "As viable water is depleted on Earth, a mission is sent to Saturn's moon Titan to retrieve sustainable H2O reserves from its alien inhabitants. But just as the humans acquire the precious resource, they are attacked by Titan rebels, who don't trust that the Earthlings will leave in peace.",
      "popularity": 250.744,
      "poster_path": "/qNz4l8UgTkD8rlqiKZ556pCJ9iO.jpg",
      "release_date": "2022-09-30",
      "title": "Attack on Titan",
      "video": false,
      "vote_average": 6,
      "vote_count": 130,
    },
    {
      "adult": false,
      "backdrop_path": "/zqkmTXzjkAgXmEWLRsY4UpTWCeo.jpg",
      "genre_ids": [
        12,
        28,
        878,
      ],
      "id": 11,
      "original_language": "en",
      "original_title": "Star Wars",
      "overview": "Princess Leia is captured and held hostage by the evil Imperial forces in their effort to take over the galactic Empire. Venturesome Luke Skywalker and dashing captain Han Solo team together with the loveable robot duo R2-D2 and C-3PO to rescue the beautiful princess and restore peace and justice in the Empire.",
      "popularity": 154.505,
      "poster_path": "/6FfCtAuVAW8XJjZ7eWeLibRLWTw.jpg",
      "release_date": "1977-05-25",
      "title": "Star Wars",
      "video": false,
      "vote_average": 8.2,
      "vote_count": 18653,
    },
    {
      "adult": false,
      "backdrop_path": "/AmR3JG1VQVxU8TfAvljUhfSFUOx.jpg",
      "genre_ids": [
        27,
        878,
      ],
      "id": 348,
      "original_language": "en",
      "original_title": "Alien",
      "overview": "During its return to the earth, commercial spaceship Nostromo intercepts a distress signal from a distant planet. When a three-member team of the crew discovers a chamber containing thousands of eggs on the planet, a creature inside one of the eggs attacks an explorer. The entire crew is unaware of the impending nightmare set to descend upon them when the alien parasite planted inside its unfortunate host is birthed.",
      "popularity": 62.433,
      "poster_path": "/vfrQk5IPloGg1v9Rzbh2Eg3VGyM.jpg",
      "release_date": "1979-05-25",
      "title": "Alien",
      "video": false,
      "vote_average": 8.1,
      "vote_count": 12802,
    },
    {
      "adult": false,
      "backdrop_path": "/jYEW5xZkZk2WTrdbMGAPFuBqbDc.jpg",
      "genre_ids": [
        878,
        12,
      ],
      "id": 438631,
      "original_language": "en",
      "original_title": "Dune",
      "overview": "Paul Atreides, a brilliant and gifted young man born into a great destiny beyond his understanding, must travel to the most dangerous planet in the universe to ensure the future of his family and his people. As malevolent forces explode into conflict over the planet's exclusive supply of the most precious resource in existence-a commodity capable of unlocking humanity's greatest potential-only those who can conquer their fear will survive.",
      "popularity": 96.469,
      "poster_path": "/d5NXSklXo0qyIYkgV94XAgMIckC.jpg",
      "release_date": "2021-09-15",
      "title": "Dune",
      "video": false,
      "vote_average": 7.8,
      "vote_count": 8876,
    },
    {
      "adult": false,
      "backdrop_path": "/8BTsTfln4jlQrLXUBquXJ0ASQy9.jpg",
      "genre_ids": [
        12,
        28,
        878,
      ],
      "id": 140607,
      "original_language": "en",
      "original_title": "Star Wars: The Force Awakens",
      "overview": "Thirty years after defeating the Galactic Empire, Han Solo and his allies face a new threat from the evil Kylo Ren and his army of Stormtroopers.",
      "popularity": 53.991,
      "poster_path": "/wqnLdwVXoBjKibFRR5U3y0aDUhs.jpg",
      "release_date": "2015-12-15",
      "title": "Star Wars: The Force Awakens",
      "video": false,
      "vote_average": 7.3,
      "vote_count": 18020,
    },
    {
      "adult": false,
      "backdrop_path": "/ziECpBRIyclmBNaFSWlvCfsKbMD.jpg",
      "genre_ids": [
        12,
        28,
        878,
      ],
      "id": 1892,
      "original_language": "en",
      "original_title": "Return of the Jedi",
      "overview": "Luke Skywalker leads a mission to rescue his friend Han Solo from the clutches of Jabba the Hutt, while the Emperor seeks to destroy the Rebellion once and for all with a second dreaded Death Star.",
      "popularity": 37.559,
      "poster_path": "/xxCnFmRZ83jHTZsBiceG4IBUGoq.jpg",
      "release_date": "1983-05-25",
      "title": "Return of the Jedi",
      "video": false,
      "vote_average": 7.9,
      "vote_count": 14223,
    },
    {
      "adult": false,
      "backdrop_path": "/SPkEiZGxq5aHWQ2Zw7AITwSEo2.jpg",
      "genre_ids": [
        12,
        28,
        878,
      ],
      "id": 181812,
      "original_language": "en",
      "original_title": "Star Wars: The Rise of Skywalker",
      "overview": "The surviving Resistance faces the First Order once again as the journey of Rey, Finn and Poe Dameron continues. With the power and knowledge of generations behind them, the final battle begins.",
      "popularity": 51.031,
      "poster_path": "/db32LaOibwEliAmSL2jjDF6oDdj.jpg",
      "release_date": "2019-12-18",
      "title": "Star Wars: The Rise of Skywalker",
      "video": false,
      "vote_average": 6.4,
      "vote_count": 8752,
    },
    {
      "adult": false,
      "backdrop_path": "/dwN0kPGrLbFRxyL3F3J3t4ShQx.jpg",
      "genre_ids": [
        12,
        14,
        28,
        53,
        878,
      ],
      "id": 18,
      "original_language": "en",
      "original_title": "The Fifth Element",
      "overview": "In 2257, a taxi driver is unintentionally given the task of saving a young girl who is part of the key that will ensure the survival of humanity.",
      "popularity": 49.34,
      "poster_path": "/fPtlCO1yQtnoLHOwKtWz7db6RGU.jpg",
      "release_date": "1997-05-02",
      "title": "The Fifth Element",
      "video": false,
      "vote_average": 7.5,
      "vote_count": 9682,
    },
    {
      "adult": false,
      "backdrop_path": "/5Iw7zQTHVRBOYpA0V6z0yypOPZh.jpg",
      "genre_ids": [
        12,
        28,
        878,
      ],
      "id": 181808,
      "original_language": "en",
      "original_title": "Star Wars: The Last Jedi",
      "overview": "Rey develops her newly discovered abilities with the guidance of Luke Skywalker, who is unsettled by the strength of her powers. Meanwhile, the Resistance prepares to do battle with the First Order.",
      "popularity": 46.743,
      "poster_path": "/kOVEVeg59E0wsnXmF9nrh6OmWII.jpg",
      "release_date": "2017-12-13",
      "title": "Star Wars: The Last Jedi",
      "video": false,
      "vote_average": 6.8,
      "vote_count": 13992,
    },
    {
      "adult": false,
      "backdrop_path": "/spHgY9GdIK6agLUnPmD0yVn3kcG.jpg",
      "genre_ids": [
        878,
        9648,
        12,
      ],
      "id": 62,
      "original_language": "en",
      "original_title": "2001: A Space Odyssey",
      "overview": "Humanity finds a mysterious object buried beneath the lunar surface and sets off to find its origins with the help of HAL 9000, the world's most advanced super computer.",
      "popularity": 36.888,
      "poster_path": "/ve72VxNqjGM69Uky4WTo2bK6rfq.jpg",
      "release_date": "1968-04-02",
      "title": "2001: A Space Odyssey",
      "video": false,
      "vote_average": 8.1,
      "vote_count": 10233,
    },
    {
      "adult": false,
      "backdrop_path": "/q7M0JpPixbEYT8EhnI7wTEMONxz.jpg",
      "genre_ids": [
        878,
        28,
        12,
      ],
      "id": 13475,
      "original_language": "en",
      "original_title": "Star Trek",
      "overview": "The fate of the galaxy rests in the hands of bitter rivals. One, James Kirk, is a delinquent, thrill-seeking Iowa farm boy. The other, Spock, a Vulcan, was raised in a logic-based society that rejects all emotion. As fiery instinct clashes with calm reason, their unlikely but powerful partnership is the only thing capable of leading their crew through unimaginable danger, boldly going where no one has gone before. The human adventure has begun again.",
      "popularity": 37.641,
      "poster_path": "/lV5OpzAss1z06YNagOVap1I35mH.jpg",
      "release_date": "2009-05-06",
      "title": "Star Trek",
      "video": false,
      "vote_average": 7.4,
      "vote_count": 9133,
    },
    {
      "adult": false,
      "backdrop_path": "/eQEYPQ98cTjLDsFthbMswQ6UDuZ.jpg",
      "genre_ids": [
        878,
        14,
        28,
      ],
      "id": 76757,
      "original_language": "en",
      "original_title": "Jupiter Ascending",
      "overview": "In a universe where human genetic material is the most precious commodity, an impoverished young Earth woman becomes the key to strategic maneuvers and internal strife within a powerful dynasty…",
      "popularity": 32.121,
      "poster_path": "/xzQ25m9vrdyvpLX74T3B3KB40Ou.jpg",
      "release_date": "2015-02-04",
      "title": "Jupiter Ascending",
      "video": false,
      "vote_average": 5.4,
      "vote_count": 6229,
    },
    {
      "adult": false,
      "backdrop_path": "/6t8ES1d12OzWyCGxBeDYLHoaDrT.jpg",
      "genre_ids": [
        28,
        12,
        878,
      ],
      "id": 330459,
      "original_language": "en",
      "original_title": "Rogue One: A Star Wars Story",
      "overview": "A rogue band of resistance fighters unite for a mission to steal the Death Star plans and bring a new hope to the galaxy.",
      "popularity": 35.143,
      "poster_path": "/i0yw1mFbB7sNGHCs7EXZPzFkdA1.jpg",
      "release_date": "2016-12-14",
      "title": "Rogue One: A Star Wars Story",
      "video": false,
      "vote_average": 7.5,
      "vote_count": 14055,
    },
    {
      "adult": false,
      "backdrop_path": "/5rYnygKCkmqWuMv1O5yAnzGF8gV.jpg",
      "genre_ids": [
        12,
        878,
        28,
      ],
      "id": 339964,
      "original_language": "en",
      "original_title": "Valerian and the City of a Thousand Planets",
      "overview": "In the 28th century, Valerian and Laureline are special operatives charged with keeping order throughout the human territories. On assignment from the Minister of Defense, the two undertake a mission to Alpha, an ever-expanding metropolis where species from across the universe have converged over centuries to share knowledge, intelligence, and cultures. At the center of Alpha is a mysterious dark force which threatens the peaceful existence of the City of a Thousand Planets, and Valerian and Laureline must race to identify the menace and safeguard not just Alpha, but the future of the universe.",
      "popularity": 33.674,
      "poster_path": "/jfIpMh79fGRqYJ6PwZLCntzgxlF.jpg",
      "release_date": "2017-07-20",
      "title": "Valerian and the City of a Thousand Planets",
      "video": false,
      "vote_average": 6.6,
      "vote_count": 6820,
    },
    {
      "adult": false,
      "backdrop_path": "/wDe8LzwuvHYYiuwyNfxdYQq8ti4.jpg",
      "genre_ids": [
        12,
        28,
        878,
      ],
      "id": 1893,
      "original_language": "en",
      "original_title": "Star Wars: Episode I - The Phantom Menace",
      "overview": "Anakin Skywalker, a young slave strong with the Force, is discovered on Tatooine. Meanwhile, the evil Sith have returned, enacting their plot for revenge against the Jedi.",
      "popularity": 34.998,
      "poster_path": "/6wkfovpn7Eq8dYNKaG5PY3q2oq6.jpg",
      "release_date": "1999-05-19",
      "title": "Star Wars: Episode I - The Phantom Menace",
      "video": false,
      "vote_average": 6.5,
      "vote_count": 13208,
    },
    {
      "adult": false,
      "backdrop_path": "/m4F1KRK5jAoQHi2mKDFE2jFKEIb.jpg",
      "genre_ids": [
        28,
        12,
        878,
      ],
      "id": 188927,
      "original_language": "en",
      "original_title": "Star Trek Beyond",
      "overview": "The USS Enterprise crew explores the furthest reaches of uncharted space, where they encounter a mysterious new enemy who puts them and everything the Federation stands for to the test.",
      "popularity": 27.7,
      "poster_path": "/cnQp8GmOWahIgQaH60Kwez3TNzw.jpg",
      "release_date": "2016-07-07",
      "title": "Star Trek Beyond",
      "video": false,
      "vote_average": 6.8,
      "vote_count": 6081,
    },
    {
      "adult": false,
      "backdrop_path": "/ojHCeDULAkQK25700fhRU75Tur2.jpg",
      "genre_ids": [
        878,
        12,
        28,
      ],
      "id": 348350,
      "original_language": "en",
      "original_title": "Solo: A Star Wars Story",
      "overview": "Through a series of daring escapades deep within a dark and dangerous criminal underworld, Han Solo meets his mighty future copilot Chewbacca and encounters the notorious gambler Lando Calrissian.",
      "popularity": 33.543,
      "poster_path": "/3IGbjc5ZC5yxim5W0sFING2kdcz.jpg",
      "release_date": "2018-05-15",
      "title": "Solo: A Star Wars Story",
      "video": false,
      "vote_average": 6.6,
      "vote_count": 7704,
    },
    {
      "adult": false,
      "backdrop_path": "/5vDuLrjJXFS9PTF7Q1xzobmYKR9.jpg",
      "genre_ids": [
        12,
        28,
        878,
      ],
      "id": 1895,
      "original_language": "en",
      "original_title": "Star Wars: Episode III - Revenge of the Sith",
      "overview": "The evil Darth Sidious enacts his final plan for unlimited power -- and the heroic Jedi Anakin Skywalker must choose a side.",
      "popularity": 36.672,
      "poster_path": "/xfSAoBEm9MNBjmlNcDYLvLSMlnq.jpg",
      "release_date": "2005-05-17",
      "title": "Star Wars: Episode III - Revenge of the Sith",
      "video": false,
      "vote_average": 7.4,
      "vote_count": 12427,
    },
    {
      "adult": false,
      "backdrop_path": "/pXnNSeyTCUebjpHTiZt7v6FZId0.jpg",
      "genre_ids": [
        12,
        28,
        878,
      ],
      "id": 1894,
      "original_language": "en",
      "original_title": "Star Wars: Episode II - Attack of the Clones",
      "overview": "Following an assassination attempt on Senator Padmé Amidala, Jedi Knights Anakin Skywalker and Obi-Wan Kenobi investigate a mysterious plot that could change the galaxy forever.",
      "popularity": 34.684,
      "poster_path": "/oZNPzxqM2s5DyVWab09NTQScDQt.jpg",
      "release_date": "2002-05-15",
      "title": "Star Wars: Episode II - Attack of the Clones",
      "video": false,
      "vote_average": 6.6,
      "vote_count": 11978,
    },
  ],
  "total_pages": 7,
  "total_results": 130,
}
`;
